toolchain/parse/typed_nodes.h

```diff
@@ -410,6 +410,28 @@ struct BuiltinFunctionDefinition {
   Lex::SemiTokenIndex token;
 };
 
+// `require` nodes
+// -------------
+using RequireIntroducer =
+    LeafNode<NodeKind::RequireIntroducer, Lex::RequireTokenIndex>;
+
+struct Impls {
+  static constexpr auto Kind = NodeKind::Impls.Define(
+      {.category = NodeCategory::Decl, .child_count = 2});
+  AnyExprId lhs;
+  Lex::ImplsTokenIndex token;
+  AnyExprId rhs;
+};
+
+struct Require {
+  static constexpr auto Kind =
+      NodeKind::Require.Define({.category = NodeCategory::Decl,
+                                .bracketed_by = RequireIntroducer::Kind});
```

Comment:
Since this is set up to require `impls`, it should have a child count of 2 instead of just bracketing.

Reply:
Should I remove `RequireIntroducer` altogether and just have the tree like this:

```
- Require
  - Impls
    - Expr
    - Expr
```

Reply:
`RequireIntroducer` should remain. For valid trees, we require every token to have a parse node (in some rare cases, two parse nodes).

Reply:
Updated